File: admin/src/main/java/com/jigumulmi/place/dto/WeeklyBusinessHourDto.java
package com.jigumulmi.place.dto;

import com.jigumulmi.place.dto.validator.ValidBusinessHour;
import java.time.DayOfWeek;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
public class WeeklyBusinessHourDto {

    @ValidBusinessHour
    private BusinessHour sunday;
    @ValidBusinessHour
    private BusinessHour monday;
    @ValidBusinessHour
    private BusinessHour tuesday;
    @ValidBusinessHour
    private BusinessHour wednesday;
    @ValidBusinessHour
    private BusinessHour thursday;
    @ValidBusinessHour
    private BusinessHour friday;
    @ValidBusinessHour
    private BusinessHour saturday;

    public BusinessHour getBusinessHour(DayOfWeek dayOfWeek) {
        return switch (dayOfWeek) {
            case DayOfWeek.SUNDAY -> sunday;
            case DayOfWeek.MONDAY -> monday;
            case DayOfWeek.TUESDAY -> tuesday;
            case DayOfWeek.WEDNESDAY -> wednesday;
            case DayOfWeek.THURSDAY -> thursday;
            case DayOfWeek.FRIDAY -> friday;
            case DayOfWeek.SATURDAY -> saturday;
        };
    }

    public void updateBusinessHour(DayOfWeek dayOfWeek, BusinessHour businessHour) {
        switch (dayOfWeek) {
            case SUNDAY -> this.sunday = businessHour;
            case MONDAY -> this.monday = businessHour;
            case TUESDAY -> this.tuesday = businessHour;
            case WEDNESDAY -> this.wednesday = businessHour;
            case THURSDAY -> this.thursday = businessHour;
            case FRIDAY -> this.friday = businessHour;
            case SATURDAY -> this.saturday = businessHour;
        }
    }

}

File: admin/src/main/kotlin/com/jigumulmi/place/dto/WeeklyBusinessHourDto.kt
package com.jigumulmi.place.dto

import com.jigumulmi.place.dto.validator.ValidBusinessHour
import java.time.DayOfWeek

data class WeeklyBusinessHourDto(
    @ValidBusinessHour
    var sunday: BusinessHour,

    @ValidBusinessHour
    var monday: BusinessHour,

    @ValidBusinessHour
    var tuesday: BusinessHour,

    @ValidBusinessHour
    var wednesday: BusinessHour,

    @ValidBusinessHour
    var thursday: BusinessHour,

    @ValidBusinessHour
    var friday: BusinessHour,

    @ValidBusinessHour
    var saturday: BusinessHour,
) {

    constructor() : this(
        BusinessHour(),
        BusinessHour(),
        BusinessHour(),
        BusinessHour(),
        BusinessHour(),
        BusinessHour(),
        BusinessHour()
    )

    fun getBusinessHour(dayOfWeek: DayOfWeek): BusinessHour {
        return when (dayOfWeek) {
            DayOfWeek.SUNDAY -> sunday
            DayOfWeek.MONDAY -> monday
            DayOfWeek.TUESDAY -> tuesday
            DayOfWeek.WEDNESDAY -> wednesday
            DayOfWeek.THURSDAY -> thursday
            DayOfWeek.FRIDAY -> friday
            DayOfWeek.SATURDAY -> saturday
        }
    }

    fun updateBusinessHour(dayOfWeek: DayOfWeek, businessHour: BusinessHour) {
        when (dayOfWeek) {
            DayOfWeek.SUNDAY -> this.sunday = businessHour
            DayOfWeek.MONDAY -> this.monday = businessHour
            DayOfWeek.TUESDAY -> this.tuesday = businessHour
            DayOfWeek.WEDNESDAY -> this.wednesday = businessHour
            DayOfWeek.THURSDAY -> this.thursday = businessHour
            DayOfWeek.FRIDAY -> this.friday = businessHour
            DayOfWeek.SATURDAY -> this.saturday = businessHour
        }
    }
}

